helix-core/src/main/java/org/apache/helix/controller/stages/ExternalViewComputeStage.java

```diff
@@ -131,20 +133,30 @@ public void process(ClusterEvent event) throws Exception {
 
       // compare the new external view with current one, set only on different
       if (curExtView == null || !curExtView.getRecord().equals(view.getRecord())) {
-        keys.add(keyBuilder.externalView(resourceName));
-        newExtViews.add(view);
+
+        // For the resource with DisableExternalView option turned on in IdealState
+        // We still compute externalView in above code for verifying and reporting cluster/resource status purpose.
+        // But we will not actually create or write the externalView to ZooKeeper.
+        if (idealState != null && idealState.isExternalViewDisabled()) {
```

Comment:
This logic will not work when idealstate is deleted. Basically idealstate will be null and in the else path we end up creating the external view. This is opposite of what we want.